src/WebJobs.Script/Binding/SendGridScriptBindingProvider.cs

```diff
@@ -1,140 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-using System;
-using System.Collections.ObjectModel;
-using System.Net.Mail;
-using System.Reflection;
-using Microsoft.Azure.WebJobs.Extensions.SendGrid;
-using Microsoft.Azure.WebJobs.Host;
-using Microsoft.Azure.WebJobs.Script.Extensibility;
-using Newtonsoft.Json.Linq;
-using SendGrid;
-using SendGrid.Helpers.Mail;
-
-namespace Microsoft.Azure.WebJobs.Script.Binding
-{
-    /// <summary>
-    /// BindingProvider for SendGrid extensions
-    /// </summary>
-    internal class SendGridScriptBindingProvider : ScriptBindingProvider
-    {
-        /// <inheritdoc/>
-        public SendGridScriptBindingProvider(JobHostConfiguration config, JObject hostMetadata, TraceWriter traceWriter)
-            : base(config, hostMetadata, traceWriter)
-        {
-        }
-
-        /// <inheritdoc/>
-        public override bool TryCreate(ScriptBindingContext context, out ScriptBinding binding)
-        {
-            if (context == null)
-            {
-                throw new ArgumentNullException("context");
-            }
-
-            binding = null;
-
-            if (string.Compare(context.Type, "sendGrid", StringComparison.OrdinalIgnoreCase) == 0)
-            {
-                binding = new SendGridBinding(context);
-            }
-
-            return binding != null;
-        }
-
-        /// <inheritdoc/>
-        public override void Initialize()
-        {
-            var sendGridConfig = CreateConfiguration(Metadata);
-            Config.UseSendGrid(sendGridConfig);
-        }
-
-        /// <inheritdoc/>
-        public override bool TryResolveAssembly(string assemblyName, out Assembly assembly)
-        {
-            assembly = null;
-
-            return Utility.TryMatchAssembly(assemblyName, typeof(SendGridAPIClient), out assembly) ||
```

Comment:
Mail client is in one of their APIs. We need to make sure (to prevent the same breaking change we had last time) that the assembly containing the extension types are shared as well.